embassy-executor/src/arch/riscv32.rs

```diff
@@ -54,20 +54,7 @@ impl Executor {
         loop {
             unsafe {
                 self.inner.poll();
-                // we do not care about race conditions between the load and store operations, interrupts
-                //will only set this value to true.
-                critical_section::with(|_| {
-                    // if there is work to do, loop back to polling
-                    // TODO can we relax this?
-                    if SIGNAL_WORK_THREAD_MODE.load(Ordering::SeqCst) {
-                        SIGNAL_WORK_THREAD_MODE.store(false, Ordering::SeqCst);
-                    }
-                    // if not, wait for interrupt
-                    else {
-                        core::arch::asm!("wfi");
-                    }
-                });
-                // if an interrupt occurred while waiting, it will be serviced here
```

Comment:
I think I break something here, but I am not sure what / how to fix.
The original code is broken if the critical section disables interrupts

Reply:
I wonder if the code should not remain the same but without the critical section, if this is a proper atomic, the critical section is not needed? I *assume* atomic instructions are emitted for the AtomicBool?

Or

Why is the store dependent on the load at all, can't the store be unconditional?

Reply:
I noticed this code is *probably* copied from xtensa which (according to the comments) ends the critical section on the wfi (equivalent) code. This is not true for RISC-V I believe